pkg/aws/route53.go

```diff
@@ -4,33 +4,80 @@ package aws
 
 import (
 	"fmt"
+	"strings"
 	"testing"
 
 	"github.com/aws/aws-sdk-go-v2/service/route53"
+	"github.com/aws/aws-sdk-go-v2/service/route53/types"
 	"github.com/stretchr/testify/assert"
 )
 
 // Route53Client is an AWS Route53 API client.
 // Typically, it's a [Route53](https://docs.aws.amazon.com/sdk-for-go/api/service/route53/#Route53).
 type Route53Client interface {
 	ListHostedZonesByNameInput(*route53.ListHostedZonesByNameInput) (*route53.ListHostedZonesOutput, error)
+	ListResourceRecordSets(*route53.ListResourceRecordSetsInput) (*route53.ListResourceRecordSetsOutput, error)
 }
 
 // AssertRoute53HostedZoneExists asserts whether or not the Route53 zone name
 // it's passed is found amongst those reported by the AWS API.
 func AssertRoute53HostedZoneExists(t *testing.T, client Route53Client, zoneName string) {
-	found := false
+	_, found, err := findZone(client, zoneName)
+
+	assert.Nil(t, err)
+	assert.True(t, found, fmt.Sprintf("'%s' not found", zoneName))
+}
+
+// AssertRoute53RecordExistsInHostedZone asserts whether or not the Route53 record
+// name it's passed exists amongst those associated with the the Route53 zone whose
+// name it's passed.
+func AssertRoute53RecordExistsInHostedZone(t *testing.T, client Route53Client, recordName string, zoneName string) {
```

Comment:
@yardbirdsax Thanks for the feedback...

> I think this would be better named as `AssertRecordExistsInHostedZone`

☝️ Good call; done!

> Can we please include a `context.Context` object as the second parameter, since this method interacts with outside resources?

☝️ I'm receptive to this but, for my own education, why do you feel this is necessary? How/where would `context.Context` be used here? (And what do you mean by "interact with outside resources" in this context?

> Should we have the input be a struct rather than explicit values

👍 Assuming IIUC, I like this. Stay tuned...

Reply:
@mdb re the `context.Context` object... the thought process here is along the lines of:
* Many libraries require a context object as part of constructors or other ops, like [ListResourceRecordSets](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/route53#Client.ListResourceRecordSets) for Route53 in the AWS SDK v2. Rather than constructing the required context in the method itself, we should allow callers to pass it in, so they can do things like set timeouts, etc.
* If methods are going to interact with outside resources, then they may take a while to do things and we may want to signal to them that we want them to stop, or we may want to have them signal us that they're done. It seems like a `Context` object is the generally thought of way to do these things, unless we want to get in to things like Channels, which seems a bit too complex.

Reply:
Yep - understood. Though, in this case, context is un-used. _Although_, it _does_ appear that `ListResourceRecordSets` expects a `Context` argument, so perhaps I incorrectly implemented the method:
https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/route53#Client.ListResourceRecordSets

Reply:
@yardbirdsax Apologies for the slow response. I believe I've implemented your suggestions. Have I conformed to the preferred `infratest` patterns?